quic/s2n-quic-platform/src/io/tokio.rs

```diff
@@ -591,24 +564,9 @@ impl<E: Endpoint<PathHandle = PathHandle>> Instance<E> {
             endpoint.transmit(&mut tx.tx_queue(), &clock);
 
             if let Some(delay) = endpoint.timeout() {
-                let delay = unsafe {
-                    // Safety: the same clock epoch is being used
-                    delay.as_duration()
-                };
-
-                // floor the delay to milliseconds to reduce timer churn
-                let delay = Duration::from_millis(delay.as_millis() as u64);
-
-                // add the delay to the clock's epoch
-                let next_time = clock.0 + delay;
-
-                // if the clock has changed let the sleep future know
-                if next_time != prev_time {
-                    sleep.as_mut().reset(next_time);
-                    prev_time = next_time;
-                }
+                timer.reset(delay);
             } else if reset_clock {
-                sleep.as_mut().reset(Instant::now() + DEFAULT_TIMEOUT);
+                timer.cancel();
```

Comment:
this doesnt match anymore. reset vs cancel. are you sure this has the same behavior?

Reply:
It doesn't have the same behavior. Before we were waking up every second when idle. When I first wrote that I figured it would be good to keep the process hot by waking up, even if we didn't have any connections. At this point, though, I think it just mostly wastes cycles. Now, if we're idle, we'll be idle.

Reply:
i meant the bool check is still called `reset_clock` but we are calling `cancel()`. There is a function called reset on the timer so its a bit confusing.

reset_clock -> cancel_clock